pkg/plugins/secret/secrets_manager/secrets_manager_test.go

```diff
@@ -0,0 +1,324 @@
+				secretPlugin := &secretsManagerSecretService{
+					client: mockSecretClient,
+				}
+				It("Should successfully store a secret", func() {
+					defer ctrl.Finish()
+
+					mockSecretClient.EXPECT().GetSecretValue(
+						gomock.Any(),
+					).Return(nil, awserr.New(secretsmanager.ErrCodeEncryptionFailure, "aws error", nil)).Times(1)
+					response, err := secretPlugin.Put(&testSecret, testSecretVal)
+					By("returning an error")
+					Expect(err).Should(HaveOccurred())
+					Expect(err.Error()).To(Equal("failed to retrieve secret: EncryptionFailure: aws error"))
+					By("returning a nil response")
+					Expect(response).Should(BeNil())
+				})
+			})
+		})
+	})
+	When("Get", func() {
+		When("Given the Secrets Manager backend is available", func() {
+			When("The secret exists", func() {
+				ctrl := gomock.NewController(GinkgoT())
+				mockSecretClient := mocks.NewMockSecretsManagerAPI(ctrl)
+				secretPlugin := &secretsManagerSecretService{
+					client: mockSecretClient,
+				}
+				It("Should return the existing secret", func() {
+					defer ctrl.Finish()
+
+					mockSecretClient.EXPECT().GetSecretValue(
+						&secretsmanager.GetSecretValueInput{
+							SecretId:  aws.String("Test"),
+							VersionId: aws.String("Version-Id"),
+						},
+					).Return(&secretsmanager.GetSecretValueOutput{
+						ARN:          aws.String(testARN),
+						Name:         aws.String("Test"),
+						VersionId:    aws.String(testVersionID),
+						SecretBinary: testSecretVal,
+					}, nil).Times(1)
+
+					response, err := secretPlugin.Access(&secret.SecretVersion{
+						Secret: &secret.Secret{
+							Name: "Test",
+						},
+						Version: "Version-Id",
+					})
+					By("Not returning an error")
+					Expect(err).ShouldNot(HaveOccurred())
+
+					By("Returning a response with the secret")
+					Expect(response).ShouldNot(BeNil())
+					Expect(response.SecretVersion.Secret.Name).Should(Equal("Test"))
+					Expect(response.SecretVersion.Version).Should(Equal("yVBWEvgpNjpcCxddXyj9kTefaUpVD999")) // Didn't return anything
+					Expect(response.Value).Should(Equal(testSecretVal))
+				})
+			})
+			When("The secret doesn't exist", func() {
+				ctrl := gomock.NewController(GinkgoT())
+				mockSecretClient := mocks.NewMockSecretsManagerAPI(ctrl)
+				secretPlugin := &secretsManagerSecretService{
+					client: mockSecretClient,
+				}
+				It("Should return a nil secret", func() {
+					defer ctrl.Finish()
+
+					mockSecretClient.EXPECT().GetSecretValue(
+						&secretsmanager.GetSecretValueInput{
+							SecretId:  aws.String("test-id"),
+							VersionId: aws.String("test-version-id"),
+						},
+					).Return(nil, fmt.Errorf("failed to get secret")).Times(1)
+
+					response, err := secretPlugin.Access(&secret.SecretVersion{
+						Secret: &secret.Secret{
+							Name: "test-id",
+						},
+						Version: "test-version-id",
+					})
+					By("Not returning an error")
+					Expect(err).Should(HaveOccurred())
```

Comment:
We probably need to document this behavior in the SDK's that an exception will be thrown secret not found.

Reply:
done